templates/myaccount/payment-methods.php

```diff
@@ -76,4 +76,6 @@
 
 <?php do_action( 'woocommerce_after_account_payment_methods', $has_methods ); ?>
 
-<a class="button" href="<?php echo esc_url( wc_get_endpoint_url( 'add-payment-method' ) ); ?>"><?php esc_html_e( 'Add payment method', 'woocommerce' ); ?></a>
+<?php if ( WC()->payment_gateways->get_available_payment_gateways() ) : ?>
```

Comment:
Perhaps add an else here to say if no gateways is available? Since the button is being hidden the customer will not be able to click through  to the add payment methods page to see the message.

Reply:
I disagree with this statement. Why would a customer expect to be able to add payment methods here? IMO we should remove any indication that this would 'normally' be possible and hide the button if not supported.